### ISSUE TITLE
See if you can block "student" from Start Livestream

### ISSUE THREAD

**Issue by NoSubstitute** [jay0lee/Meet-Extensions#1]
First of all, THANK YOU! :-)
The features you have seem to work just fine.
Student can't Mute nor Kick.

But, if the teacher also has enabled Livestream, and there are students in the regular Meet, then they can _start the livestream_. Here are a couple of images of what it looks like. Excuse the Swedish.

I am confident that you can find the code to master that too.

![Screenshot 2020-03-17 at 12 01 11](https://user-images.githubusercontent.com/12572734/76851214-32a81900-6849-11ea-9054-5f4c8bd5c720.png)
![Screenshot 2020-03-17 at 12 01 21](https://user-images.githubusercontent.com/12572734/76851216-33d94600-6849-11ea-9a6f-afa95feb19ee.png)
![Screenshot 2020-03-17 at 12 01 38](https://user-images.githubusercontent.com/12572734/76851219-3471dc80-6849-11ea-8dc0-4707fe15c473.png)
![Screenshot 2020-03-17 at 12 02 04](https://user-images.githubusercontent.com/12572734/76851311-579c8c00-6849-11ea-86bd-9bad748e1384.png)

**Comment by NoSubstitute** [comment]
It's possible to start the Livestream from two places.
Link in top view and in the three-dots-menu.

**Comment by jay0lee** [comment]
Livestream setting is only available if teacher enabled it on calendar
event. It requires intentional action by teacher.

Typically the idea would be to give students *only* the Livestream link,
not the hangout meeting link or both, why give them both?

So I assume this is primarily about student starting Livestream early?

**Comment by NoSubstitute** [comment]
The teacher may very well have students both in the classroom and at home, and may want to give either an opportunity to participate with screenshare and/or voice/video. Meaning the teacher invites them to the regular Meet, but don't want them to start/stop the streaming.

**Comment by jay0lee** [comment]
What is the point in enabling livestream if all students will be in regular
meeting? Again if teacher doesn't enable livestream on regular calendar
invite then there should be no option to enable livestream within the
meeting.

Also need to point out that this is proof of concept code. Please don't
deploy it to live students. Despite basically working to block student
mute/kick it has not been heavily tested and may cause more classroom
disruption that it solves.

Jay Lee

**Comment by NoSubstitute** [comment]
Yes, I took the warning to heart. Only testing with my own test user so
far. Will grow test group slowly.

Hmm, maybe I was unclear. I think there are several occasions where the
teacher would want a *few* students inside the Meet and the rest only on
Livestream. As you say, there's not much point if *all* students are inside
Meet, unless they are streaming to another audience.

It appears all participants inside a Meet see the *Start streaming* button,
and if it could be blocked in a similar way that would be grand.

Now, I haven't yet tested if disallowing students from Streaming, in admin
settings, affects their ability to see the button. I will test that, as
that may be an alternative solution.
